### DISCUSSION TITLE
Ant Media Server logs rotation policy.

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5657] [FAQs]
I would like to understand how the rotation policy works for the ant-media-server.log and antmedia-error.log files.
Are they rotated on the basis of the log file size or number of days/time-period?
Also, how can I change these policies for both the ant-media-server.log and antmedia-error.log files?

Thank you

**Comment by USAMAWIZARD** [comment]
I will check this and get back to you

**Reply by Mohit-3196** [reply]
Hi @USAMAWIZARD,
No worries. @muratugureminoglu abi will guide us on this topic.

**Comment by muratugureminoglu** [comment]
Hi @Mohit-3196 

There are different configurations for the `antmedia-error.log` and `ant-media-server.log` files. Let me explain them to you.

**ant-media-server.log**: 
When the file size limit reaches 50MB, log rotation starts, and the log is compressed and stored as `ant-media-server.1.log.gz`. A maximum of 3 log files is retained.

You can change these settings in the following file:

```
/usr/local/antmedia/conf/logback.xml
```

**Log file size:** [Logback.xml Line 26](https://github.com/ant-media/Ant-Media-Server/blob/master/src/main/server/conf/logback.xml#L26)
**Number of log files retained:** [Logback.xml Lines 22-23](https://github.com/ant-media/Ant-Media-Server/blob/master/src/main/server/conf/logback.xml#L22-L23)

After making these changes, don't forget to restart Ant Media Server:

```
systemctl restart antmedia
```

**antmedia-error.log and 0.0.0.0_access.*.log**:
We use the logrotate tool for these log files. The files are retained for 7 days, and the maximum file size is limited to 50MB.

You can make changes in the following file:

```
/etc/logrotate.d/antmedia
```
**Log file size:** maxsize 50M
**Number of log files retained:** rotate 7
 
https://github.com/ant-media/Scripts/blob/master/install_ant-media-server.sh#L423-L450

No need to restart for this process.

If you still have any questions or if anything is unclear, please feel free to ask.

**Reply by Mohit-3196** [reply]
Thank you.
